### DISCUSSION TITLE
Having some challenges building OAuth/OIDC support for use in fluent-bit

### DISCUSSION THREAD

**Discussion by jmcshane** [edenhill/librdkafka#3657] [Q&A]
At this point, librdkafka-1.7.0 is referenced directly [within fluent-bit](https://github.com/fluent/fluent-bit/blob/master/plugins/out_kafka/librdkafka-1.7.0/), so none of the `oidc` properties are available. I attempted to compile in support by replacing that directory with the current master branch code and recompiling (using the `-DWITH_CURL` and `-DWITH_SASL_OAUTHBEARER` flags). The build does successfully add the `rdkafka_sasl_oauthbearer_oidc.c` to the compile list, but it does not recognize that libcurl is available to the build process:

```
[2021/12/20 20:38:39] [error] [output:kafka:kafka.0] cannot configure 'sasl.oauthbearer.client.secret' property: 'Configuration property "sasl.oauthbearer.client.secret" not supported in this build: OAuth/OIDC depends on libcurl which was not available at build time'
```

I am running on a rhel7 based docker image, installing `libcurl-devel` via yum before running `cmake`. I understand that this is a bit of an odd issue because the build is sourced from the fluent-bit repo itself, but the challenge here is the detection of libcurl support at compile time. My cmake command for fluent-bit is:

```
RUN cmake  \
          -DFLB_HTTP_SERVER=On \
          -DFLB_IN_SYSTEMD=On \
          -DWITH_SSL=On \
          -DWITH_CURL=On \
          -DWITH_SASL=On \
          -DWITH_SASL_OAUTHBEARER=On \
          -DFLB_OUT_KAFKA=On ..
```

**Comment by edenhill** [comment]
librdkafka's cmake builds are unsupported and are maintained by the community. It currently lacks curl/oidc support. PRs welcome.

Recommend using ./configure and make
